### PR TITLE
- add yasm

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **yasm/1.3.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/yasm/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (Croydon) [inline recipes/yasm/all/conanfile.py]: please remove this two meta lines

File: recipes/yasm/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import os
+
+
+class YASMInstallerConan(ConanFile):
+    name = "yasm"
+    version = "1.3.0"
+    url = "https://github.com/bincrafters/conan-yasm_installer"
+    description = "Yasm is a complete rewrite of the NASM assembler under the “new” BSD License"
+    license = "BSD"
+    exports_sources = ["LICENSE"]
```
Comment (Croydon) [inline recipes/yasm/all/conanfile.py]: remove this line

File: recipes/yasm/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+# -*- coding: utf-8 -*-
+
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, MSBuild
+import os
+
+
+class YASMInstallerConan(ConanFile):
+    name = "yasm"
+    version = "1.3.0"
+    url = "https://github.com/bincrafters/conan-yasm_installer"
```
Comment (Croydon) [inline recipes/yasm/all/conanfile.py]:
```suggestion
    url = "https://github.com/conan-io/conan-center-index"
```

File: recipes/yasm/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = 'yasm-%s' % self.version
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.download('https://raw.githubusercontent.com/yasm/yasm/master/YASM-VERSION-GEN.bat',
```
Comment (Croydon) [inline recipes/yasm/all/conanfile.py]: Why is here an extra download happening? shouldn't be all relevant files be already in the download archive?

Even if not, downloading from master seems definitely like a bad idea

Comment (SSE4) [reply recipes/yasm/all/conanfile.py]: the file is missing from the released archive. I may change it to use specific revision/tag (or alternatively, just put file near to the recipe?)

File: recipes/yasm/all/conanfile.py (hunk 2 above)
Comment (uilianries) [inline recipes/yasm/all/conanfile.py]:
```suggestion
```

File: recipes/yasm/all/conanfile.py (hunk 4 above)
Comment (uilianries) [reply recipes/yasm/all/conanfile.py]: Yes, please, use a the commit sha1. You can use this one, which is the latest:

https://github.com/yasm/yasm/blob/bcc01c59d8196f857989e6ae718458c296ca20e3/YASM-VERSION-GEN.bat

File: recipes/yasm/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from conans import ConanFile
+import os
+
+
+class TestPackage(ConanFile):
+
+    def test(self):
+        self.run("yasm --help")
```
Comment (uilianries) [inline recipes/yasm/all/test_package/conanfile.py]: I'm curious if run_environment is required here not, I mean, what is the chance of finding yasm from system first.

Comment (SSE4) [reply recipes/yasm/all/test_package/conanfile.py]: it doesn't use any shared libraries, executable is standalone, but I may add it, just for consistency

File: recipes/yasm/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/yasm/all/conanfile.py]: maybe we can add automated checks into hook for this and others?
it's pretty boring to do all these adjustments after PR every time.
my process is:
- copy recipe from bincrafters
- add `conandata.yml` and `config.yml`
- fix any errors conan center hook reported
as a contributor, after that, I would expect an approval, as soon as CI passes.

Comment (danimtb) [reply recipes/yasm/all/conanfile.py]: We are still polishing the conan center hook and this first PRs help us with the task. If you encounter repetitive things like this that are not blocked by the Hook, please open an issue in the repo to improve it and help with the onboarding of future people to conan center index. Thanks!

**Pull request merged**